### PR TITLE
Rework kotlin generator with kotlin-poet

### PR DESCRIPTION
**Pull request by sokomishalov**
PR goal is to decrease plain string manipulations to build kotlin API code and provide more safety while generating code.

Make sure that:

- [x] You have read the [contribution guidelines](https://github.com/lettuce-io/lettuce-core/blob/main/.github/CONTRIBUTING.md).
- [ ] You have created a feature request first to discuss your contribution intent. Please reference the feature request ticket number in the pull request.
- [x] You use the code formatters provided [here](https://github.com/lettuce-io/lettuce-core/blob/main/formatting.xml) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -665,7 +673,7 @@
                 <artifactId>kotlin-maven-plugin</artifactId>
                 <configuration>
                     <jvmTarget>1.8</jvmTarget>
-                    <languageVersion>1.3</languageVersion>
+                    <languageVersion>1.5</languageVersion>
```
Comment (sokomishalov) [inline pom.xml:676]: this one is needed to support trailing commas in kotlin generated code

**Comment by mp911de** [comment]
Thanks for the context. Given that Kotlinpoet doesn't seem to want to improve their dev experience and is rather limiting, I start to dislike the entire move into Kotlinpoet.

**Comment by sokomishalov** [comment]
JFYI fixed indentation, total diff decreased significantly and now is quite more readable

**Comment by sokomishalov** [comment]
Let's keep our generator as is for now

**Pull request closed without merging**